tools/src/main.rs

```diff
@@ -334,3 +344,37 @@ fn generate_residue_g2_elements(vk: &HashMap<String, Value>) -> String {
 
     residue_g2_elements
 }
+
+
+fn get_modexp_function(l2_mode: bool) -> String {
```

Comment:
Very nit with low priority: I would want to make foundry fuzz test to catch that these two functions behave exactly the same way.

Reply:
Okay, I'll create a task for that